### DISCUSSION TITLE
Why so many .exe files?

### DISCUSSION THREAD

**Discussion by ghost** [Cuperino/QPrompt-Teleprompter#244] [General]
![qprompt](https://github.com/Cuperino/QPrompt-Teleprompter/assets/36171926/87fb4f00-958f-4af9-aaa3-8b422750ca6d)
 
Hi there, just curious as to why there are so many .exe files?

**Comment by Cuperino** [comment]
Hi @siliconunit, 

Good question. Those files come from dependencies of the Qt framework and of the KDE libraries used to make QPrompt. They're introduced to the build by [Craft](https://community.kde.org/Craft), the pipeline software I've used to create the .EXE installer, amongst other builds.

Craft was created by a member of the KDE Community, and is maintained by her and other members of it. Craft is used to package all Windows, Mac, Linux AppImage, and Android versions of [KDE software](https://apps.kde.org/). Given my few interactions with them, and contribution attempts, I have no reason to mistrust Craft developers or think of them as the source of a supply chain attack.

QPrompt's development development is tied to KDE software in the sense that, as of current and early access versions, it makes use of various KDE libraries and tools. Nevertheless, QPrompt is not a KDE Project because I don't want to adhere to the [KDE Manifesto](https://manifesto.kde.org/) for the time being, because I want the freedom to experiment and succeed with commercialization options before proposing that QPrompt joins the KDE Community. This means that, although the KDE community has always been very welcoming, I haven't received much help or any code contributions when it's come to making use of their tooling. In other words, I'm just figuring things out on my own, not entirely understanding what I'm doing.

As I said in the first paragraph, those executables come from dependencies of the Qt framework and of KDE libraries. I suspect they're not all needed. `xmllint` for example is a XML code linting tool that's coming from Qt; I don't see why that would be needed at runtime, but it might be used for parsing the HTML, given HTML is a superset of XML, but that communication would normally be done via a .dll on Windows, instead of an .exe; the exe is more of a developer tool. Sqllite is another Qt dependency which I don't use at all in QPrompt, so it should be removed altogether. Gettext comes from GNU, and is brought in by KDE's Ki18n. I don't know how Ki18n is implemented or whether gettext's exe interface is needed; again it's the sort of thing I'd expect to be done via a library.

QPrompt has a custom Craft repository, where I specify which dependencies to include, and which to leave out when making these builds. You can inspect it and use it to produce your own builds, it is found here: [https://github.com/Cuperino/craft-blueprints-cuperino](https://github.com/Cuperino/craft-blueprints-cuperino). Detailed build instructions are over here: [https://github.com/Cuperino/QPrompt-Teleprompter/blob/main/build.md#kdes-craft-build-system](https://github.com/Cuperino/QPrompt-Teleprompter/blob/main/build.md#kdes-craft-build-system).

If you have time and would like to improve on the Windows installer, contributions are most welcome. Nevertheless, if you read what I've written in QPrompt's blog, you'll notice that I'm a little frustrated with Craft, and that [our pipeline is currently broken, and has been so for a few months](https://www.patreon.com/posts/build-delayed-92362735).

[Instead of fixing Craft I would prefer for us to create a new installer using CPack and NSIS.](https://www.patreon.com/posts/qprompt-towards-95558451) Craft is still our plan B, so either direction we go would be good. Before attempting to make a new installer, I'm trying to make a new build pipeline to source our dependencies from. These can be found in the `vcpkg` and `vcpkg-kirigami` branches. If you'd like to contribute, all help is welcomed. In the end, all I want is to spend less time having to fix build pipelines that break, and more time to develop QPrompt. If we can [get QPrompt to most platforms in the process](https://github.com/Cuperino/QPrompt-Teleprompter/discussions/121), that would be great!

Thank you for reaching out, and I hope I've answered your question!

**Comment by Cuperino** [comment]
Closing discussion because of inactivity. Feel free to re-open.
